packages/access-api/src/routes/validate-email.js

```diff
@@ -25,7 +31,11 @@ export async function validateEmail(req, env) {
       return new HtmlResponse(
         (
           <ValidateEmail
-            delegation={delegation}
+            email={delegation.capabilities[0].nb.identity.replace(
```

Comment:
It could be good to define a utility function for extracting the email from the ability, because this logic using `String#replace` won't work for `did:mailto` using the syntax we changed to once realizing `@` was not a valid DID character.

Reply:
and then it could be used on line 84 below as well

Reply:
after reading more, I also don't really understand where `nb.identity` is coming from or what type it would be (I don't see `nb.identity` in the access capabilities). If I'm reading right, it might be type `any` when the `replace` method is called on it here. We might avoid runtime bugs by clarifying what the return type of `env.models.validations.put` is expecting (e.g. specific `Capability` types within the delegation, not generic allowing any `Capabilities` tuple) and/or doing more explicit parsing of `req.query.ucan` instead of using the type assertion on line 26

Reply:
(it would be good to add a test for what should happen if someone tries to inject malicious `req.query.ucan` and IMO it's best to ensure we don't create db records without heavy parsing)

Reply:
this is from the old recover cap it will go away soon, regarding ucan validation i agree we should validate the caps for the specific mode and the signature. 


https://github.com/web3-storage/w3protocol/issues/411